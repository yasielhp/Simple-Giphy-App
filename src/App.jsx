import { useState } from "react"
import { AddCategory, GifGrid, Header } from "./components/"
import { Container } from "@nextui-org/react";


export const App = () => {
  const [categories, setCategories] = useState(['trending'])

  const handleAddCategory = (newCategory) => {
    if(categories.includes(newCategory)) return
    setCategories([newCategory, ...categories])
  }
  return (
    <Container md>
      <Header/>
      <AddCategory onNewCategory={e => handleAddCategory(e)}/>
         { categories.map((category) => (<GifGrid key={category} category={category } />))}
    </Container>
  )
}
