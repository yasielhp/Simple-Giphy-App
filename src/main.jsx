import {createRoot} from 'react-dom/client'
import { App } from './App'
import { NextUIProvider, createTheme } from '@nextui-org/react';
import { ThemeProvider as NextThemesProvider } from 'next-themes';


const lightTheme = createTheme({
  type: 'light',
})

const darkTheme = createTheme({
  type: 'dark',
})

const rootElement = document.getElementById("root");
const root = createRoot(rootElement);

root.render(
  <NextThemesProvider
    defaultTheme="system"
    attribute="class"
    value={{
      light: lightTheme.className,
      dark: darkTheme.className
    }}
  >
    <NextUIProvider>
      <App />
    </NextUIProvider>
  </NextThemesProvider>
);

