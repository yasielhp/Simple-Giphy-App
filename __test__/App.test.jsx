import { App } from '../src/App';
import { render,screen } from '@testing-library/react';

describe('Test in <App/>', () => {
  test('- Debe de hacer match con el snapshot', () => {
    render(<App />);
    const { container } = screen
    expect(container).toMatchSnapshot()
   })
})