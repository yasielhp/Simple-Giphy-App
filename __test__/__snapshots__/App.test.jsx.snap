// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <App/> - Debe de hacer match con el snapshot 1`] = `undefined`;
